File: app/src/main/java/com/billpayments/features/saved_billers/data/SavedBiller.kt
package com.billpayments.features.saved_billers.data

data class SavedBiller(
    val billerId: String,
    val account: String,
    val nickname: String,
)

data class SavedBillers(val items: List<SavedBiller>) {

    fun contains(billerId: String, account: String): Boolean =
        items.any { it.billerId == billerId && it.account == account }

    fun adding(saved: SavedBiller): SavedBillers = SavedBillers(items + saved)

    fun removing(billerId: String, account: String): SavedBillers =
        SavedBillers(items.filterNot { it.billerId == billerId && it.account == account })

    companion object {
        val EMPTY = SavedBillers(emptyList())
    }
}

File: BillPayments/Features/SavedBillers/Data/SavedBiller.swift
struct SavedBiller: Codable, Equatable {
    let billerId: String
    let account: String
    let nickname: String
}

struct SavedBillers: Equatable {
    let items: [SavedBiller]

    func contains(billerId: String, account: String) -> Bool {
        items.contains { $0.billerId == billerId && $0.account == account }
    }

    func adding(_ saved: SavedBiller) -> SavedBillers {
        SavedBillers(items: items + [saved])
    }

    func removing(billerId: String, account: String) -> SavedBillers {
        SavedBillers(items: items.filter { !($0.billerId == billerId && $0.account == account) })
    }

    static let empty = SavedBillers(items: [])
}
